src/zarr/abc/metadata.py

```diff
@@ -44,3 +44,6 @@ def from_dict(cls, data: dict[str, JSON]) -> Self:
         ...
 
         return cls(**data)
+
+
+__all__ = ["Metadata"]
```

Comment:
I find it odd that all the new __all__ are at the bottom of files - I think more discoverable if they're at the top after imports but before function/class definitions (that's also how I've seen them everywhere else)?